code/parachain/frame/cosmwasm/src/instrument.rs

```diff
@@ -31,3 +36,299 @@ pub fn gas_and_stack_instrumentation(
+			Instruction::I64RemU |
+			Instruction::I32RemU |
+			Instruction::I64RemS |
+			Instruction::I32RemS => self.i64rems,
+			Instruction::I64And | Instruction::I32And => self.i64and,
+			Instruction::I64Or | Instruction::I32Or => self.i64or,
+			Instruction::I64Xor | Instruction::I32Xor => self.i64xor,
+			Instruction::I64Shl | Instruction::I32Shl => self.i64shl,
+			Instruction::I64ShrU |
+			Instruction::I32ShrU |
+			Instruction::I64ShrS |
+			Instruction::I32ShrS => self.i64shrs,
+			// TODO(aeryz): I64Sub
+			Instruction::I64Rotl | Instruction::I32Rotl => self.i64rotl,
+			Instruction::I64Rotr | Instruction::I32Rotr => self.i64rotr,
+			Instruction::I32WrapI64 => self.i32wrapi64,
+			Instruction::I64ExtendSI32 | Instruction::I64ExtendUI32 => self.i64extendsi32,
+			Instruction::F64Eq | Instruction::F32Eq => self.f64eq,
+			Instruction::F64Ne | Instruction::F32Ne => self.f64ne,
+			Instruction::F64Lt | Instruction::F32Lt => self.f64lt,
+			Instruction::F64Gt | Instruction::F32Gt => self.f64gt,
+			Instruction::F64Le | Instruction::F32Le => self.f64le,
+			Instruction::F64Ge | Instruction::F32Ge => self.f64ge,
+			Instruction::F64Abs | Instruction::F32Abs => self.f64abs,
+			Instruction::F64Neg | Instruction::F32Neg => self.f64neg,
+			Instruction::F64Ceil | Instruction::F32Ceil => self.f64ceil,
+			Instruction::F64Floor | Instruction::F32Floor => self.f64floor,
+			Instruction::F64Trunc | Instruction::F32Trunc => self.f64trunc,
+			Instruction::F64Nearest | Instruction::F32Nearest => self.f64nearest,
+			Instruction::F64Sqrt | Instruction::F32Sqrt => self.f64sqrt,
+			Instruction::F64Add | Instruction::F32Add => self.f64add,
+			Instruction::F64Sub | Instruction::F32Sub => self.f64sub,
+			Instruction::F64Mul | Instruction::F32Mul => self.f64mul,
+			Instruction::F64Div | Instruction::F32Div => self.f64div,
+			Instruction::F64Min | Instruction::F32Min => self.f64min,
+			Instruction::F64Max | Instruction::F32Max => self.f64max,
+			Instruction::F64Copysign | Instruction::F32Copysign => self.f64copysign,
+			Instruction::Drop => self.i64const,
+			Instruction::Select => self.select,
+			Instruction::If(_) => self.if_,
+			Instruction::Else => self.else_,
+			Instruction::GetLocal(_) => self.getlocal,
+			Instruction::SetLocal(_) => self.setlocal,
+			Instruction::TeeLocal(_) => self.teelocal,
+			Instruction::GetGlobal(_) => self.getglobal,
+			Instruction::SetGlobal(_) => self.setglobal,
+			Instruction::CurrentMemory(_) => self.currentmemory,
+			Instruction::GrowMemory(_) => self.growmemory,
+			Instruction::Br(_) => self.br,
+			Instruction::BrIf(_) => self.brif,
+			Instruction::BrTable(table) => self
+				.brtable
+				.saturating_add(self.brtable_per_elem.saturating_mul(table.table.len() as u32)),
+			Instruction::Call(_) => self.call,
+			// TODO(aeryz): CallIndirect will be implemented after the pallet is upgraded to IBC
+			// compatible VM
+			Instruction::CallIndirect(_, _) => self.call,
```

Comment:
I don't understand the requirement for IBC here, could you explain :)

Reply:
I need to change code_gen but the IBC changes are merged to main. We can either wait for IBC pr in pallet side, or in this PR, I can implement that functions by returning Error::Unsupported